pkg/services/object/put/distributed.go

```diff
@@ -138,11 +152,30 @@ func (t *distributedTarget) sendObject(node nodeDesc) error {
 
 	target := t.nodeTargetInitializer(node)
 
-	if err := target.WriteObject(t.obj, t.objMeta, t.encodedObject); err != nil {
+	err := target.WriteObject(t.obj, t.objMeta, t.encodedObject)
+	if err != nil {
 		return fmt.Errorf("could not write header: %w", err)
-	} else if _, err := target.Close(); err != nil {
+	}
+
+	_, sig, err := target.Close()
+	if err != nil {
 		return fmt.Errorf("could not close object stream: %w", err)
 	}
+
+	if !node.local {
```

Comment:
shouldnt this be
```suggestion
	if t.localNodeInContainer && !node.local {
```
? Seems like ur proposal would fail when request goes through a non-container node

Reply:
hm, that's interesting. what a non-contianer node should do if it faces an object whose meta signatures it should check? @roman-khimov

Reply:
Why would it be involved in replication?

Reply:
what does a node do when it is asked to sign with session key objects of a container it does not belong to?

Reply:
It's signing and passing the object to container node.

Reply:
oh, i have overthought some things in this code but yes, `t.localNodeInContainer` is enough, added. (also added nil signature check)